backend/controller/controller.go

```diff
@@ -98,10 +98,14 @@ type Service struct {
 	deploymentReservationTimeout time.Duration
 	artefactChunkSize            int
 	key                          model.ControllerKey
+	ctx                          context.Context
```

Comment:
I added this `ctx` here because the sink is processing logs in the background and the request `ctx` seems to get canceled before I get a chance to write the logs to the db. So I'm using the `controller`'s context for the sink. Maybe there's a better way?

Reply:
Yes, you can use `unstoppable.Context(ctx)`.